src/api/director.rs

```diff
@@ -113,30 +113,43 @@ pub struct TufUpdates {
 
 impl TufUpdates {
     /// Convert `TargetRequests` to `TufUpdates`.
-    pub fn from(requests: TargetRequests) -> Self {
-        Self {
-            targets: requests.requests.into_iter().map(|(id, req)| (id, Self::to_update(req))).collect(),
-        }
+    pub fn from(requests: TargetRequests) -> Result<Self> {
+        Ok(Self {
+            targets: requests
+                .requests
+                .into_iter()
+                .map(|(id, req)| Ok((id, Self::to_update(req)?)))
+                .collect::<Result<HashMap<_, _>>>()?,
```

Comment:
What is the HashMap for?

Reply:
So while parsing the TOML file, we build a hashmap from `HardwareId` to a `TargetRequest` as this represents the layout of the TOML file that we want. Director expects a different layout of course, so this `from` function converts these `TargetRequests` into `TufUpdates`.

The map step simply passes the hardware ID unchanged (saving one entire string copy!), and the final step will collect each into a new hashmap entry from `HardwareId` to `TufUpdate`.